### PR TITLE
system: use UTC dates and ISO 8601 Z form

### PR DESCRIPTION
**Pull request by jbenito3**
* Created new database type to store UTC-like dates
* Validated that the date is ISO 8601
* Refactored tests and added extra tests for restful api.

### CODE REVIEW

**Comment by jbenito3** [comment]
@tiborsimko please, pay special attention to the test refactoring I did. Thanks!

**Comment by lnielsen** [comment]
FYI, https://sqlalchemy-utils.readthedocs.org/en/latest/data_types.html#module-sqlalchemy_utils.types.arrow might be useful as well

**Comment by jbenito3** [comment]
@tiborsimko now the tests have passed for travis...but they don't in my computer...Shall we merge?

**Comment by jbenito3** [comment]
@lnielsen yes, that could be a nice alternative. I copied the idea from our Indico Friends, [see here](https://github.com/indico/indico/blob/8940eba6ac7913832499ac49527e2126d7c95bf3/indico/core/db/sqlalchemy/custom/utcdatetime.py#L21). I thought it would be easier to use the original types with a simple decorator and avoid adding one extra lib.

**Comment by tiborsimko** [comment]
> now the tests have passed for travis...but they don't in my computer...

Why don't we skip the conflicting import statements in `test_restful_api.py`:

```python
from webtest import TestApp   # isort:skip

from .base import BaseTest  # isort:skip
```

This should make both local tests and Travis CI happy.

File: .travis.yml [hunk 1]
```diff
@@ -47,10 +53,11 @@ before_install:
 install:
   - travis_retry pip install --upgrade pip
   - travis_retry pip install coveralls
-  - travis_retry pip install pytest pytest-pep8 pytest-pep257 pytest-cov pytest-cache pytest-isort
+  - travis_retry pip install --upgrade pytest pytest-pep8 pytest-pep257 pytest-cov pytest-cache pytest-isort
   - travis_retry pip install -e .
```
Comment (tiborsimko) [inline .travis.yml]: The `pip --upgrade` is perhaps not necessary?  (see also my previous `isort:skip` message)

File: .travis.yml [hunk 2]
```diff
@@ -28,6 +28,12 @@ language: python
 python:
   - "3.4"
 
+addons:
+  postgresql: "9.4"
+
+env:
+  - SQLALCHEMY_DATABASE_URI=postgres://postgres@localhost:5432/claims
+
```
Comment (tiborsimko) [inline .travis.yml]: If we change the database name, let's use the same name in `docker-compose.yml` too?  

BTW `claimstore` may be more appropriate DB name, with `claims` being one of its tables.

File: claimstore/modules/claims/models.py [hunk 3]
```diff
@@ -42,9 +44,8 @@ class Claim(db.Model):
         default=lambda: str(uuid4())
     )
     created = db.Column(
-        db.DateTime,
-        nullable=False,
-        default=db.func.current_timestamp()
+        UTCDateTime,
+        nullable=False
     )
```
Comment (tiborsimko) [inline claimstore/modules/claims/models.py]: Nobody is really using ClaimStore much yet, so this comment is for the future, just to say to think about future Alembic based DB structure upgrade recipes.

File: claimstore/static/json/examples/claim.inspire.1.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   },
   "claim": {
     "predicate": "is_same_as",
-    "datetime": "2015-05-25 11:00:00 UTC",
+    "datetime": "2015-05-25T11:00:00Z",
     "certainty": 80,
```
Comment (tiborsimko) [inline claimstore/static/json/examples/claim.inspire.1.json]: Can you please also amends ISO Z dates in `docs/index.rst`?

File: tests/base.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+# USA.
+
+"""Base test class."""
+
+from unittest import TestCase
+
+from claimstore.app import create_app, db
+
+
+class BaseTest(TestCase):
```
Comment (tiborsimko) [inline tests/base.py]: A `FooTestCase` class name, such as `BaseTestCase` or `ClaimStoreTestCase`, might better express relation to upstream `TestCase`?

File: tests/test_restful_api.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+"""claimstore.modules.claims.restful test suite."""
+
+import json
+import os
+
+import pytest
+
+from webtest import TestApp
+
```
Comment (tiborsimko) [inline tests/test_restful_api.py]: Pylon's `WebTest` looks nice, I did not know about it.  An alternative would be to use `Flask-Testing` e.g. as in [our Flask-IIIF test suite](https://github.com/inveniosoftware/flask-iiif/tree/master/tests), but we can definitely stay with `WebTest`.

File: .travis.yml (hunk 1 above)
Comment (jbenito3) [inline .travis.yml]: Right, I will remove it and I will try `isort:skip`.

File: claimstore/modules/claims/models.py (hunk 3 above)
Comment (jbenito3) [inline claimstore/modules/claims/models.py]: Yes sure, Alembic was in my mind for production days.

**Comment by jbenito3** [comment]
@tiborsimko Thank you for your comments. I have done all the modifications as you suggested.

**Pull request merged**